azkaban-web-server/src/main/java/azkaban/imagemgmt/services/ImageVersionServiceImpl.java

```diff
@@ -72,6 +77,32 @@ public int createImageVersion(ImageMetadataRequest imageMetadataRequest)
   @Override
   public List<ImageVersion> findImageVersions(ImageMetadataRequest imageMetadataRequest)
       throws ImageMgmtException {
-    return imageVersionsDao.findImageVersions(imageMetadataRequest);
+    Set<String> imageTypes = new LinkedHashSet<>();
+    imageTypes.add("azkaban_core");
+    imageTypes.add("azkaban_exec");
+    imageTypes.add("hadoop_java_job");
+    imageTypes.add("kabootar_job");
+    imageTypes.add("wormhole_job");
+    return imageVersionsDao.getActiveVersionByImageTypes(imageTypes);
+    //return imageVersionsDao.findImageVersions(imageMetadataRequest);
+  }
+
+  @Override
+  public void updateImageVersion(ImageMetadataRequest imageMetadataRequest) throws IOException,
+      ImageMgmtException {
+    // Convert input json payload to image version update request object
+    ImageVersionRequest imageVersionRequest =
+        converterUtils.convertToModel(imageMetadataRequest.getJsonPayload(),
+            ImageVersionRequest.class);
```

Comment:
Let me provide more context here. The idea of using this structure is to handle complete business logic and processing at service layer. ImageMetadataRequest is designed as generic DTO it encapsulates the input and passes them between layers. Here are some of the use cases and usefulness of the approach.
1. In the servlet layer there is no business logic as such. Predominantly it reads parameters and input payload and encapsulates them in ImageMetadataRequest and passes them to the service layer.
2. Service layer methods are designed to accept ImageMetadataRequest and it serves most of the use cases such as create, update, and search with multiple parameters etc.
3. The service layer designed to do most of the things such as input payload to model conversion, model validation and other business logic etc. This structure also helps write some useful unit test on service layer as most of he business logic is in one place.
4. In case of create and update the respective model conversion (from input payload) happens as at the service layer. In case of search request ImageMetadataRequest is passed to DAO layer as it is. The search parameters are extracted from ImageMetadataRequest and search query can be built using that. Moreover, we know the type of query parameters in ImageMetadataRequest and they can be converted to the respective type and used to build the query. So ImageMetadataRequest is a multi-purpose DTO.
5.By passing single DTO i.e. ImageMetadataRequest in the service layer method we need not change the signature of the signature of the methods if new parameters are added and deleted.
6. Again there is no hard and fast rule to convert user input (payload) only at the Servlet layer. In fact, it helps to writes some validation or tests as majority of logic is present in service layer.